functions/nodejs-apps.bash

```diff
@@ -202,3 +202,147 @@ nodered_setup() {
+    if ! cond_redirect sudo -u "${username:-openhabian}" git checkout HEAD -- npm-shrinkwrap.json; then echo "FAILED git"; fi
+    if ! cond_redirect sudo -u "${username:-openhabian}" git pull; then echo "FAILED git"; fi
+    if ! cond_redirect sudo -u "${username:-openhabian}" npm install zigbee2mqtt ; then echo "FAILED npm"; fi
+    if ! cond_redirect sudo -u "${username:-openhabian}" cp -R data-backup/* data; then echo "FAILED (cp backup)"; fi
+    rm -rf data-backup
+    cd /opt || echo "FAILED (cd)"
+    if ! cond_redirect systemctl start zigbee2mqtt; then echo "FAILED"; fi
+
+    if [[ -n $INTERACTIVE ]]; then
+      whiptail --title "Operation successful" --msgbox "$updateSuccessText" 15 80
+    fi
+    echo "OK"
+    return 0
+  fi
+
+  # get usb adapters for radio menu
+  while IFS= read -r line; do
+    ((i++))
+    my_adapters="$my_adapters $i $line $loopSel "
+    adapterArray+=("$line")
+    loopSel=0
+  done < <( ls /dev/serial/by-id )
+  
+  # ask for user input parameters
+  if [[ -n $INTERACTIVE ]]; then
+    if ! (whiptail --title "Zigbee2MQTT installation" --yes-button "Continue" --no-button "Cancel" --yesno "$introText" 14 80); then echo "CANCELED"; return 0; fi
+    if ! selectedAdapter=$(whiptail --notags --title "Zigbee2MQTT installation" --radiolist "$adapterText" 14 100 4 $my_adapters 3>&1 1>&2 2>&3); then return 0; fi
+    if ! mqttUser=$(whiptail --title "MQTT User" --inputbox "$mqttUserText" 10 80 "$mqttDefaultUser" 3>&1 1>&2 2>&3); then return 0; fi
+    if ! mqttPW=$(whiptail --title "MQTT password" --passwordbox "$mqttPWText" 10 80 3>&1 1>&2 2>&3); then return 0; fi
+    if ! (whiptail --title "Zigbee2MQTT installation" --infobox "$installText" 14 80); then echo "CANCELED"; return 0; fi
+  fi
+  if [[ -z $mqttUser ]] ; then mqttUser="$mqttDefaultUser"; fi
+
+  if ! node_is_installed || is_armv6l; then
+    echo -n "$(timestamp) [openHABian] Installing zigbee2mqtt prerequsites (NodeJS)... "
+    if cond_redirect nodejs_setup; then true; else echo "FAILED"; return 1; fi
+  fi
+
+  echo -n "$(timestamp) [openHABian] Downloading zigbee2mqtt... "
+  zigbee2mqttBase="$(npm list | head -n 1)/node_modules/zigbee2mqtt"
+  if [[ -d $zigbee2mqttBase ]]; then
+    if cond_redirect systemctl stop zigbee2mqtt.service; then echo "OK (stop service)"; else echo "FAILED (stop service)"; return 1; fi # Stop the service 
+    cond_echo "Removing any old installations..."
+    cond_redirect npm uninstall zigbee2mqtt
+  fi
+  if ! cond_redirect zigbee2mqtt_download "/opt"; then echo "FAILED (download)"; return 1; fi
+  echo "OK"
+
+  echo -n "$(timestamp) [openHABian] Creating log directory... "
+  mkdir  /var/log/zigbee2mqtt || (echo "FAILED (create log-directory)"; return 1)
+  chown openhabian /var/log/zigbee2mqtt || (echo "FAILED (create log-directory)"; return 1)
+  chgrp openhab /var/log/zigbee2mqtt || (echo "FAILED (create log-directory)"; return 1)
```

Comment:
Logging is an annoying territory, make sure that you add these permissions to the permissions corrections function to allow users to easily fix them if they get messed up. This is especially important for users who run zram.

Reply:
ok. I hope I did that correct...